src/server/game/Entities/Player/Player.cpp

```diff
@@ -2031,6 +2033,39 @@ void Player::RegenerateAll()
     }
 
     m_regenTimer = 0;
+
+    // Handles the emotes for drinking and eating.
+    // According to sniffs there is a background timer going on that repeats independed from the time window where the aura applies.
+    // That's why we dont need to reset the timer on apply. In sniffs I have seen that the first call for the spell visual is totally random, then after
+    // 5 seconds over and over again which confirms my theory that we have a independed timer.
+    if (m_foodEmoteTimerCount >= 5000)
+    {
+        std::vector<Aura*> auraList;
+        AuraEffectList const& ModRegenAuras = GetAuraEffectsByType(SPELL_AURA_MOD_REGEN);
+        AuraEffectList const& ModPowerRegenAuras = GetAuraEffectsByType(SPELL_AURA_MOD_POWER_REGEN);
+
+        for (auto itr = ModRegenAuras.begin(); itr != ModRegenAuras.end(); ++itr)
```

Comment:
Please! you know exactly how much elements your vector will have, seems like a place to use reserve.

Also, you can use vector::insert to copy list elements without for

Reply:
Not necessarily. There may be more regeneration affecting auras aside from drinking food. such as buffs.

Reply:
In how far? At this point we don't know how many auras we will get. We might be just getting one aura. We might also get 10 or 20.
Also I used emplace_back to get a sorted list to prevent random behaivior between food and drinking auras.